src/Account.sol

```diff
@@ -69,64 +61,102 @@ contract MarginBase is MinimalProxyable, IMarginBase, OpsReady {
     /// @param value: value to check if zero
     modifier notZero(uint256 value, bytes32 valueName) {
         /// @notice value cannot be zero
-        if (value == 0) {
-            revert ValueCannotBeZero(valueName);
-        }
+        if (value == 0) revert ValueCannotBeZero(valueName);
+
         _;
     }
 
     /*//////////////////////////////////////////////////////////////
                               CONSTRUCTOR
     //////////////////////////////////////////////////////////////*/
 
-    /// @notice constructor never used except for first CREATE
-    constructor() MinimalProxyable() {}
+    /// @notice disable initializers on initial contract deployment
+    /// @dev set owner of implementation to zero address
+    constructor() Owned(address(0)) {
+        // recommended to use this to lock implementation contracts
+        // that are designed to be called through proxies
+        _disableInitializers();
+    }
 
     /// @notice allows ETH to be deposited directly into a margin account
     /// @notice ETH can be withdrawn
     receive() external payable onlyOwner {}
 
-    /// @notice initialize contract (only once) and transfer ownership to caller
+    /// @notice initialize contract (only once) and transfer ownership to specified address
     /// @dev ensure resolver and sUSD addresses are set to their proxies and not implementations
+    /// @param _owner: account owner
     /// @param _marginAsset: token contract address used for account margin
     /// @param _addressResolver: contract address for Synthetix address resolver
-    /// @param _marginBaseSettings: contract address for MarginBase account settings
+    /// @param _settings: contract address for account settings
     /// @param _ops: gelato ops address
+    /// @param _factory: contract address for account factory
     function initialize(
+        address _owner,
         address _marginAsset,
         address _addressResolver,
-        address _marginBaseSettings,
-        address payable _ops
-    ) external initOnce {
+        address _settings,
+        address payable _ops,
+        address _factory
+    ) external initializer {
+        owner = _owner;
+        emit OwnershipTransferred(address(0), _owner);
+
         marginAsset = IERC20(_marginAsset);
         addressResolver = IAddressResolver(_addressResolver);
+
+        /// @dev Settings must exist prior to account creation
+        settings = ISettings(_settings);
```

Comment:
How do we update settings, say if we add new functionality to settings that's needed on the upgraded accounts?

Reply:
Settings is an isolated, deployed once (until updating) contract that the account implementation points to. We can roll out a system upgrade with a new settings address if we wish to upgrade it. Given it will only ever store fee-related state, it does not matter that its storage is not looking at the proxy.

Reply:
As far as updating fees or other things, as long as there is a setter, it is simply doing that as the owner

Reply:
I will write a test doing this btw

Reply:
Yeah a test for this would be good

Reply:
🏁🏁🏁